frontend/src/views/Transfers/ViewTransfer/ViewTransfer.jsx

```diff
@@ -1,46 +1,48 @@
 // hooks
-import { useNavigate, useParams } from 'react-router-dom'
-import { useMemo } from 'react'
-import { useTransfer } from '@/hooks/useTransfer'
-import { decimalFormatter } from '@/utils/formatters'
-import { useCurrentUser } from '@/hooks/useCurrentUser'
-import { useTranslation } from 'react-i18next'
-import { roles } from '@/constants/roles'
-// Components
-import {
-  Typography,
-  Stepper,
-  Step,
-  StepLabel,
-  Stack,
-  List,
-  ListItem
-} from '@mui/material'
 import BCBox from '@/components/BCBox'
 import BCButton from '@/components/BCButton'
+import BCModal from '@/components/BCModal'
 import Loading from '@/components/Loading'
 import { Role } from '@/components/Role'
-// Icons
-import SyncAltIcon from '@mui/icons-material/SyncAlt'
+import { roles } from '@/constants/roles'
+import { TRANSACTIONS } from '@/constants/routes/routes'
+import { useCurrentUser } from '@/hooks/useCurrentUser'
+import { useHasRoles } from '@/hooks/useRole'
+import { useTransfer, useUpdateTransfer } from '@/hooks/useTransfer'
+import { decimalFormatter } from '@/utils/formatters'
+
+import { faArrowLeft, faCircle } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import SyncAltIcon from '@mui/icons-material/SyncAlt'
 import {
-  faCircle,
-  faArrowLeft,
-  faTrash
-} from '@fortawesome/free-solid-svg-icons'
-// Sub components
-import { OrganizationBadge } from '../../Transactions/components/OrganizationBadge'
-import { demoData } from '../../Transactions/components/demo'
+  List,
+  ListItem,
+  Stack,
+  Step,
+  StepLabel,
+  Stepper,
+  Typography
+} from '@mui/material'
+import { useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate, useParams } from 'react-router-dom'
+
 import { AttachmentList } from '../../Transactions/components/AttachmentList'
 import { Comments } from '../../Transactions/components/Comments'
+import { OrganizationBadge } from '../../Transactions/components/OrganizationBadge'
+import { demoData } from '../../Transactions/components/demo'
+import { rescindButton } from '../buttonConfigs'
 
 export const ViewTransfer = () => {
+  const [modalData, setModalData] = useState(null)
   const { t } = useTranslation(['common', 'transfer'])
   const iconSizeStyle = {
     fontSize: (theme) => `${theme.spacing(12)} !important`
   }
   const navigate = useNavigate()
   const { transferId } = useParams()
+  const hasRoles = useHasRoles([roles.transfers, roles.signing_authority])
```

Comment:
wouldn't this restrict us from using any other roles? Say for certain content to display we want admin roles and for others, supplier roles. But using this would make us to declare multiple hasRoles variables like hasAdminRole, hasTransferRole?

Reply:
yea thats true, if we are creating components or functions that need to cater to multiple users with different roles this wont work.

We would need to specify roles within the component/function with conditionals to account for different users.

Any ideas on how we might be able to improve this?

Reply:
actually, the `useCurrentUser` hook has a `hasRoles` function that can replace this. that way we can use that function where ever we need